sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -299,33 +299,35 @@ case class Elt(children: Seq[Expression])
       """
     }
 
-    val cases = ctx.buildCodeBlocks(assignStringValue)
-    val codes = if (cases.length == 1) {
-      s"""
-        UTF8String $stringVal = null;
-        switch ($indexVal) {
-          ${cases.head}
-        }
-       """
-    } else {
-      var prevFunc = "null"
-      for (c <- cases.reverse) {
-        val funcName = ctx.freshName("eltFunc")
-        val funcBody = s"""
-         private UTF8String $funcName(InternalRow ${ctx.INPUT_ROW}, int $indexVal) {
-           UTF8String $stringVal = null;
-           switch ($indexVal) {
-             $c
-             default:
-               return $prevFunc;
-           }
-           return $stringVal;
-         }
-        """
-        val fullFuncName = ctx.addNewFunction(funcName, funcBody)
-        prevFunc = s"$fullFuncName(${ctx.INPUT_ROW}, $indexVal)"
-      }
-      s"UTF8String $stringVal = $prevFunc;"
+    var prevFunc = "null"
+    var codes = ctx.splitExpressionsWithCurrentInputs(
+      expressions = assignStringValue,
+      funcName = "eltFunc",
+      extraArguments = ("int", indexVal) :: Nil,
+      returnType = "UTF8String",
+      makeSplitFunction = body =>
+        s"""
+           |UTF8String $stringVal = null;
+           |switch ($indexVal) {
+           |  $body
+           |  default:
+           |    return $prevFunc;
+           |}
+           |return $stringVal;
+        """.stripMargin,
+      foldFunctions = funcs => s"UTF8String $stringVal = ${funcs.last};",
+      makeFunctionCallback = f => prevFunc = s"$f(${ctx.INPUT_ROW}, $indexVal)",
+      mergeSplit = false)
```

Comment:
We don't need to and can't merge split functions in inner classes.

We don't need to to do it because the split functions are not call in a sequence like this:
```
eltFunc_1(...)
eltFunc_2(...)
...
```

The calls are embedded in the default branch in each split function. So we won't call all split inner functions in outer class.

Reply:
We can't merge them because the `makeSplitFunction` will create invalid merged function if used with the given `foldFunctions`:

```java
private UTF8String eltFunc(InternalRow i, int index) { 

  UTF8String stringVal = null;
  switch (index) {
    UTF8String stringVal = eltFunc_999(i, index);
    default:
      return nestedClassInstance.eltFunc_999(i, index);
  }
  return stringVal;
}
```

Reply:
yes but in this way we can hit the 64KB limit. Moreover I think that the current implementation is quite complex. What about making it similar to any other implementations using a `while` loop instead of a `switch`? WDYT?

Reply:
Why we can hit the 64kb limit?

Reply:
I have thought about it. Other implementation needs to introduce at least one global variable such as case when case. If we can tolerate it, it is ok for me. Let's see what other reviewers think about it.

Reply:
let's not complicated the already-complex `splitExpressions`, I'm ok to use some global variables to simplify the code.

Reply:
@viirya I think we can hit it, with an outstanding number of parameters to the function. I am not saying that it is likely to happen, but IMHO it is feasible to make it happening

Reply:
Ok. Let me replace it with simpler codes. Thanks.